src/wallet/gtest/test_wallet.cpp

```diff
@@ -0,0 +1,444 @@
+    CWalletTx wtx;
+    EXPECT_EQ(0, wtx.mapNoteData.size());
+
+    mapNoteData_t noteData;
+    auto sk = libzcash::SpendingKey::random();
+    JSOutPoint jsoutpt {wtx.GetTxid(), 0, 1};
+    CNoteData nd {sk.address(), uint256()};
+    noteData[jsoutpt] = nd;
+
+    EXPECT_THROW(wtx.SetNoteData(noteData), std::runtime_error);
+}
+
+TEST(wallet_tests, find_note_in_tx) {
+    CWallet wallet;
+
+    auto sk = libzcash::SpendingKey::random();
+    wallet.AddSpendingKey(sk);
+
+    auto wtx = GetValidReceive(sk, 10, true);
+    auto note = GetNote(sk, wtx, 0, 1);
+    auto nullifier = note.nullifier(sk);
+
+    auto noteMap = wallet.FindMyNotes(wtx);
+    EXPECT_EQ(1, noteMap.size());
+
+    JSOutPoint jsoutpt {wtx.GetTxid(), 0, 1};
+    CNoteData nd {sk.address(), nullifier};
+    EXPECT_EQ(1, noteMap.count(jsoutpt));
+    EXPECT_EQ(nd, noteMap[jsoutpt]);
+}
+
+TEST(wallet_tests, get_conflicted_notes) {
+    CWallet wallet;
+
+    auto sk = libzcash::SpendingKey::random();
+    wallet.AddSpendingKey(sk);
+
+    auto wtx = GetValidReceive(sk, 10, true);
+    auto note = GetNote(sk, wtx, 0, 1);
+    auto nullifier = note.nullifier(sk);
+
+    auto wtx2 = GetValidSpend(sk, note, 5);
+    auto wtx3 = GetValidSpend(sk, note, 10);
+    auto hash2 = wtx2.GetTxid();
+    auto hash3 = wtx3.GetTxid();
+
+    // No conflicts for no spends
+    EXPECT_EQ(0, wallet.GetConflicts(hash2).size());
+    wallet.AddToWallet(wtx, true, NULL);
+    EXPECT_EQ(0, wallet.GetConflicts(hash2).size());
+
+    // No conflicts for one spend
+    wallet.AddToWallet(wtx2, true, NULL);
+    EXPECT_EQ(0, wallet.GetConflicts(hash2).size());
+
+    // Conflicts for two spends
+    wallet.AddToWallet(wtx3, true, NULL);
+    auto c3 = wallet.GetConflicts(hash2);
+    EXPECT_EQ(2, c3.size());
+    EXPECT_EQ(std::set<uint256>({hash2, hash3}), c3);
+}
+
+TEST(wallet_tests, nullifier_is_spent) {
+    CWallet wallet;
+
+    auto sk = libzcash::SpendingKey::random();
+    wallet.AddSpendingKey(sk);
+
+    auto wtx = GetValidReceive(sk, 10, true);
+    auto note = GetNote(sk, wtx, 0, 1);
+    auto nullifier = note.nullifier(sk);
+
+    EXPECT_FALSE(wallet.IsSpent(nullifier));
+
+    wallet.AddToWallet(wtx, true, NULL);
+    EXPECT_FALSE(wallet.IsSpent(nullifier));
+
+    auto wtx2 = GetValidSpend(sk, note, 5);
+    wallet.AddToWallet(wtx2, true, NULL);
+    EXPECT_FALSE(wallet.IsSpent(nullifier));
+
+    // Fake-mine the transaction
+    EXPECT_EQ(-1, chainActive.Height());
+    CBlock block;
+    block.vtx.push_back(wtx2);
+    block.hashMerkleRoot = block.BuildMerkleTree();
+    auto blockHash = block.GetHash();
+    CBlockIndex fakeIndex {block};
+    mapBlockIndex.insert(std::make_pair(blockHash, &fakeIndex));
+    chainActive.SetTip(&fakeIndex);
+    EXPECT_TRUE(chainActive.Contains(&fakeIndex));
+    EXPECT_EQ(0, chainActive.Height());
+
+    wtx2.SetMerkleBranch(block);
+    wallet.AddToWallet(wtx2, true, NULL);
+    EXPECT_TRUE(wallet.IsSpent(nullifier));
+
+    // Tear down
```

Comment:
The way to do this "properly" is to write a test case class and the use it for the test. However, this would put this test into a different test case than all the other tests in this file, meaning that test filtering would not work as expected. I'm not sure if it's possible to have test case classes linked to existing test case names. #1324 might also have an effect here.